From c50d295c22066e95e526129dabc4f4a7da975896 Mon Sep 17 00:00:00 2001
From: Henry Robinson <henry@apache.org>
Date: Fri, 30 Mar 2012 01:01:35 +0000
Subject: [PATCH 07/19] ZOOKEEPER-1435 cap space usage of default log4j rolling policy (phunt via henryr)

git-svn-id: https://svn.apache.org/repos/asf/zookeeper/trunk@1307190 13f79535-47bb-0310-9956-ffa450edef68
(cherry picked from commit abc3d9dd7e1ec00fced27d4d5734cf3fd407c8bc)
---
 conf/log4j.properties |   55 +++++++++++++++++++++++++++---------------------
 1 files changed, 31 insertions(+), 24 deletions(-)

diff --git a/conf/log4j.properties b/conf/log4j.properties
index 9825970..61cf575 100644
--- a/conf/log4j.properties
+++ b/conf/log4j.properties
@@ -1,29 +1,40 @@
+# Copyright 2012 The Apache Software Foundation
+# 
+# Licensed to the Apache Software Foundation (ASF) under one
+# or more contributor license agreements.  See the NOTICE file
+# distributed with this work for additional information
+# regarding copyright ownership.  The ASF licenses this file
+# to you under the Apache License, Version 2.0 (the
+# "License"); you may not use this file except in compliance
+# with the License.  You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
 # Define some default values that can be overridden by system properties
 zookeeper.root.logger=INFO, CONSOLE
+
 zookeeper.console.threshold=INFO
+
 zookeeper.log.dir=.
 zookeeper.log.file=zookeeper.log
-zookeeper.log.threshold=DEBUG
+zookeeper.log.threshold=INFO
+zookeeper.log.maxfilesize=256MB
+zookeeper.log.maxbackupindex=20
+
 zookeeper.tracelog.dir=.
 zookeeper.tracelog.file=zookeeper_trace.log
 
-#
-# ZooKeeper Logging Configuration
-#
-
-# Format is "<default threshold> (, <appender>)+
-
-# DEFAULT: console appender only
 log4j.rootLogger=${zookeeper.root.logger}
 
-# Example with rolling log file
-#log4j.rootLogger=DEBUG, CONSOLE, ROLLINGFILE
-
-# Example with rolling log file and tracing
-#log4j.rootLogger=TRACE, CONSOLE, ROLLINGFILE, TRACEFILE
-
 #
-# Log INFO level and above messages to the console
+# console
+# Add "console" to rootlogger above if you want to use this 
 #
 log4j.appender.CONSOLE=org.apache.log4j.ConsoleAppender
 log4j.appender.CONSOLE.Threshold=${zookeeper.console.threshold}
@@ -32,23 +43,19 @@ log4j.appender.CONSOLE.layout.ConversionPattern=%d{ISO8601} [myid:%X{myid}] - %-
 
 #
 # Add ROLLINGFILE to rootLogger to get log file output
-#    Log DEBUG level and above messages to a log file
+#
 log4j.appender.ROLLINGFILE=org.apache.log4j.RollingFileAppender
 log4j.appender.ROLLINGFILE.Threshold=${zookeeper.log.threshold}
 log4j.appender.ROLLINGFILE.File=${zookeeper.log.dir}/${zookeeper.log.file}
-
-# Max log file size of 10MB
-log4j.appender.ROLLINGFILE.MaxFileSize=10MB
-# uncomment the next line to limit number of backup files
-#log4j.appender.ROLLINGFILE.MaxBackupIndex=10
-
+log4j.appender.ROLLINGFILE.MaxFileSize=${zookeeper.log.maxfilesize}
+log4j.appender.ROLLINGFILE.MaxBackupIndex=${zookeeper.log.maxbackupindex}
 log4j.appender.ROLLINGFILE.layout=org.apache.log4j.PatternLayout
 log4j.appender.ROLLINGFILE.layout.ConversionPattern=%d{ISO8601} [myid:%X{myid}] - %-5p [%t:%C{1}@%L] - %m%n
 
-
 #
 # Add TRACEFILE to rootLogger to get log file output
-#    Log DEBUG level and above messages to a log file
+#    Log TRACE level and above messages to a log file
+#
 log4j.appender.TRACEFILE=org.apache.log4j.FileAppender
 log4j.appender.TRACEFILE.Threshold=TRACE
 log4j.appender.TRACEFILE.File=${zookeeper.tracelog.dir}/${zookeeper.tracelog.file}
-- 
1.7.0.4

